Login

gharamorth opened the pull request.

gharamorth committed:
Created the possibility to see all current questionnaires (basic)

gharamorth committed:
Created several viewmodels, views, added them onto the locator. You can now
-see all questionnaires
-view a questionnaire
-open a view to add a new question to the viewed questionnaire

gharamorth committed:
SelectedQuestionType can now properly be selected (changes in the CreateQuestionViewModel) and now properly display their names.

gharamorth committed:
I'm not sure either.

gharamorth committed:
Primary merge commit

gharamorth committed:
Implemented basic login functionality
Notes:
-The password box is not a true password box, it's a textbox.
-If the login is succesful it will be saved in the HomeViewModel's new Account property
-The loginview does not yet automatically close.

gharamorth committed:
Undoing overriding commit (please work)

gharamorth committed:
Resolving merge issues (accepted remote all)

gharamorth committed:
Merging vraag_aanmaken_revert and development

gharamorth committed:
Merging of development with vraag_aanmaken_revert

gharamorth committed:
Open vraag kan nu succesvol aangemaakt worden
Note: Alleen open vragen momenteel mogelijk ivm XAML complicaties

gharamorth committed:
Removed test variables

gharamorth committed:
Beginnings changing the code to the CRUD-oriented implementation

gharamorth committed:
Minor bugs fixed

gharamorth committed:
Login system should now be ready for test.

Evert-Arends commented on FSBeheer/FSBeheer/FSContext.cs:
```diff
@@ -8,6 +8,8 @@ namespace FSBeheer
 
     public partial class FSContext : DbContext
     {
+        internal object temp;
```
Delete?

Evert-Arends replied:
@gharamorth Is dit opgelost?

Evert-Arends commented on FSBeheer/FSBeheer/ViewModel/AccountVM.cs:
```diff
@@ -0,0 +1,15 @@
+
+namespace FSBeheer.ViewModel
+{
+    public class AccountVM : ViewModelBase
```
Do not

Evert-Arends commented on FSBeheer/FSBeheer/ViewModel/CreateQuestionViewModel.cs:
```diff
@@ -0,0 +1,69 @@
+
+            _question = new Question();
+
+            using (var context = new CustomFSContext())
```
Please use the correct classes for this functionality as stated in the CustomFXContext

Evert-Arends commented on FSBeheer/FSBeheer/ViewModel/CreateQuestionViewModel.cs:
```diff
@@ -0,0 +1,69 @@
+            }
+        }
+
+        public void AddQuestion()
```
What?

gharamorth replied:
What

Evert-Arends commented on FSBeheer/FSBeheer/ViewModel/LoginViewModel.cs:
```diff
@@ -0,0 +1,108 @@
+            string username;
+            string password;
+            string salt;
+            int i = 5;
```
Please remove

gharamorth committed:
Ensured that existing accounts in the seeder use encrypted passwords

gharamorth committed:
Linebreaks and such.

gharamorth committed:
commit cause I want to switch branches

gharamorth committed:
All question types now possible, if not efficiently so.

gharamorth committed:
Questions can now be created, the user must be aware of the syntax when inserting new options. When fields that do not align with the selected question types are filled in, their values will not be inserted into the database.

gharamorth committed:
Switch commit

gharamorth committed:
swtiching

gharamorth committed:
Resolving pull request issues

gharamorth committed:
Removing unnecessary code AccountVM

gharamorth committed:
Merging vraag_aanmaken_revert and login

gharamorth committed:
Merging of login and vraag_aanmaken, resolving issues surrounding the login.

gharamorth committed:
Fixen merge conflicts bij het aanmaken van een vraag
Notes:
-deze branch bevat een view voor questionnaires (list) en questionnaire (edit)
-Sommige onderdelen van de commentaar zijn outdated
-Kleine bug bij het toevoegen van commentaar bij het aanmaken van een nieuwe vraag

gharamorth committed:
Refactoring ViewModel<-> VM responsibility separation

gharamorth committed:
Merge of login&vraag_aanmaken_revert with development

gharamorth committed:
Fixed merging issues regarding questionnaire
To do: login repositioning, adding related properties to homeviewmodel

gharamorth committed:
Manually merged existing duplicate views/viewmodels

Evert-Arends commented on FSBeheer/FSBeheer/VM/QuestionnaireVM.cs:
```diff
@@ -1,119 +1,70 @@
+using System;
 
-namespace FSBeheer.VM
+namespace FSBeheer.ViewModel
```
Wrong namespace? This should be in the VM folder? Please explain a bit further?

gharamorth replied:
Wrong namespace yeah, it is in the VM folder

gharamorth committed:
Changed QuestionnaireVM namespace FSBeheer.ViewModel to FSBeheer.VM

Evert-Arends commented:
Even development pullen en pushen, dan merge ik m

gharamorth committed:
Merged (accepted remote)

gharamorth committed:
Accepted remote

gharamorth committed:
Merge with development

gharamorth committed:
Added missing usings breaking the build.

Evert-Arends commented on FSBeheer/FSBeheer/ViewModel/QuestionnairesViewModel.cs:
```diff
@@ -0,0 +1,56 @@
+
+        private QuestionnaireEditView _questionnaireEditView;
+
+        public ICommand ViewQuestionnaireCommand { get; set; }
```
ICommand should not be used, use RelayCommand instad.

gharamorth committed:
Removed QuestionnairesViewModel as it has been replaced with QuestionnairesListViewModel and fixed Phi's account seed in the seeder (it wasn't hashed and wouldn't have worked with the current login system);
